### PR TITLE
Deprecate imagestreamtag acceptance

### PR DESCRIPTION
**Pull request by bradmwilliams**
This PR contains the following changes:

1. `release-controller-api`
    - Derive the release phase status from the `ReleasePayload` conditions instead of annotations on the release `imagestreamtag` 
    - Replace `release.openshift.io/message` annotation reads with `PayloadOverride.Reason` from the `ReleasePayload` spec

2. `release-tool.py`: 
    - Remove the legacy `imagestreamtag` patching path from  — accept/reject now exclusively patches the `ReleasePayload`

3. `release-payload-controller`:
    - Fix a race condition in `ComputeJobState()` that could prematurely reject a release while blocking jobs still had retries outstanding

## Summary by CodeRabbit

* **New Features**
  * Added an admin mode to the release tool for elevated accept/reject actions.
* **Bug Fixes**
  * Improved phase/status handling across release listings, details, and dashboards by basing resolution on release payload state.
  * Corrected failing/rejection evaluation and link/alert behavior when phases are unknown or job/aggregate states are empty.
* **Tests**
  * Expanded unit tests for release phase derivation and job state computation, including unknown/empty edge cases.

### CODE REVIEW

File: hack/release-tool.py [hunk 1]
```diff
@@ -650,12 +578,12 @@ def keep(ctx, namespace, imagestream, release, delete, execute):
...
                             logger.info(f' - {namespace}/{tag.name()}')
 
-        except (ValueError, OpenShiftPythonException, Exception) as e:
+        except (OpenShiftPythonException, Exception) as e:
             logger.error(f'Unable to process imagestreamtag: "{namespace}/{imagestream}:{release}"')
             raise e
 
 
-def approval(ctx, namespace, imagestream, release, team, accept, reject, delete, check, execute):
+def approval(ctx, namespace, release, team, accept, reject, delete, check, execute):
```
[the remark this answers is not in the document]
Comment (bradmwilliams) [reply hack/release-tool.py]: Done!

File: hack/release-tool.py [hunk 2]
```diff
@@ -929,7 +860,7 @@ def annotate_imagestream(ctx, namespace, name, annotations, execute):
                 exit(1)
             team = args['check_team']
             check = True
-        approval(context, release_namespace, release_image_stream, args['release'][0], team, accept, reject, delete, check, args['execute'])
+        approval(context, options, release_namespace, release_image_stream, args['release'][0], team, accept, reject, delete, check, args['execute'])
```
[the remark this answers is not in the document]
Comment (bradmwilliams) [reply hack/release-tool.py]: Done!

**Review by AlexNPavel — APPROVED** [review]
